pyiron_workflow/channels.py

```diff
@@ -777,7 +783,42 @@ def __init__(
                 object.
         """
         super().__init__(label=label, node=node)
-        self.callback: callable = callback
+        if self._is_node_method(callback) and self._takes_zero_arguments(callback):
+            self._callback: str = callback.__name__
+        else:
+            raise BadCallbackError(
+                f"The channel {self.label} on {self.node.label} got an unexpected "
+                f"callback: {callback}. "
+                f"Lives on node: {self._is_node_method(callback)}; "
+                f"take no args: {self._takes_zero_arguments(callback)} "
+            )
+
+    def _is_node_method(self, callback):
+        try:
+            return callback == getattr(self.node, callback.__name__)
+        except AttributeError:
+            return False
+
+    def _takes_zero_arguments(self, callback):
+        return callable(callback) and self._no_positional_args(callback)
+
+    @staticmethod
+    def _no_positional_args(func):
+        return (
+            sum(
+                1
+                for parameter in inspect.signature(func).parameters.values()
+                if (
+                    parameter.default == inspect.Parameter.empty
+                    and parameter.kind != inspect._ParameterKind.VAR_KEYWORD
+                )
+            )
+            == 0
+        )
```

Comment:
```suggestion
        for parameter in inspect.signature(func).parameters.values():
            if parameter.default == inspect.Parameter.empty and parameter.kind != inspect._ParameterKind.VAR_KEYWORD:
                return False
        return True
```

Reply:
Nice improvement-- Early stopping and more readable. Looking at this, I think we can do it with an `all()` comprehension and further get rid of the branching return!

Reply:
I can commit this (so you're in the history here) and do that tomorrow, but if you see this in time feel free to modify the suggestion and directly commit it. (Assuming you like the idea)

Reply:
Ah I see. I like the idea